chat_shell/chat_shell/messages/attachment_preview.py

```
@@ -0,0 +1,324 @@
+    body_budget = max(0, total_limit - reserved)
+
+    body_sizes = [counter.count_text(seg_body) for _, seg_body in headers_and_bodies]
+    allocations = _allocate_budget(body_sizes, body_budget)
+
+    rebuilt: list[str] = []
+    truncated_segments = 0
+    before_tokens = 0
+    after_tokens = 0
+    for (header, seg_body), alloc, attachment_id, orig_tokens in zip(
+        headers_and_bodies, allocations, ids, body_sizes
+    ):
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)